mgmt/rest/task.go

```diff
@@ -95,7 +96,7 @@ func (s *Server) addTask(w http.ResponseWriter, r *http.Request, _ httprouter.Pa
 	if tr.Name != "" {
 		opts = append(opts, core.SetTaskName(tr.Name))
 	}
-	opts = append(opts, core.OptionStopOnFailure(10))
+	opts = append(opts, core.OptionStopOnFailure(tr.MaxFailures))
```

Comment:
We need a way to handle the case where MaxFailures is not set and supply a default. I believe you can just check to see if tr.MaxFailures is equal to 0, if it is we can supple the default, if not we pass through the supplied value. 

This seems to be the root cause of the test failures since this particular test does not use the client library.